### PR TITLE
Delete HACK_NO_XPCALL and related code

### PR DESCRIPTION
**Pull request by MagiMaster**
Not 100% sure how this is used right now. I don't see any uses in the app codebase, but perhaps people use it during local testing? If so, it might be safer to deprecate/stub this instead of deleting it outright.

### CODE REVIEW

**Comment by MagiMaster** [comment]
Ah. Looks like Studio uses it (e.g. DraftsWidget/src/TestHelpers/MockDraftsService.spec.lua) so this change isn't quite safe yet.

**Comment by LPGhatguy** [comment]
I think it would be better to stub, since this is used in a few places, like you noted.

**Comment by MagiMaster** [comment]
I checked with Arseny and he said the flags for yieldable xpcall are already removed, so it should be safe to depend on it.

File: src/TestPlanner.lua [hunk 1]
```diff
@@ -42,132 +42,6 @@ local function buildPlan(builder, module, env)
...
-
-	env.step = env.it
-
-	env.fit = env.itFOCUS
-	env.xit = env.itSKIP
-	env.fdescribe = env.describeFOCUS
-	env.xdescribe = env.describeSKIP
-
-	return env
-end
```
Comment (LPGhatguy) [inline src/TestPlanner.lua]: Was this dead code that we forgot about during the previous env cleanup?

Comment (MagiMaster) [reply src/TestPlanner.lua]: I think that's just from having these on separate branches. I don't see that anymore.

**Pull request merged**